python/src/py_stream/py_egress_stream.rs

```diff
@@ -0,0 +1,66 @@
+use erdos::dataflow::stream::{errors::TryReadError, EgressStream};
+use pyo3::{exceptions, prelude::*};
+
+use crate::PyMessage;
+
+/// The internal Python abstraction over an `EgressStream`.
+///
+/// This class is exposed on the Python interface as `erdos.streams.EgressStream`.
+#[pyclass]
+pub struct PyEgressStream {
+    egress_stream: EgressStream<Vec<u8>>,
```

Comment:
It's kinda non-intuitive as to why `PyEgressStream` is not inheriting from `PyStream`, especially so because `EgressStream`'s Python interface extends the `Stream` Python ABC. We should probably remove the inheritance in Python since Rust `EgressStream`s don't implement the `Stream` trait either.

Reply:
Agreed, `EgressStream` shouldn't inherit the `Stream` ABC.